discojs/src/aggregator/byzantine.ts

```diff
@@ -0,0 +1,185 @@
+import createDebug from "debug";
+import { List, Map } from "immutable";
+
+import { AggregationStep, Aggregator } from "./aggregator.js";
+import { WeightsContainer, client } from "../index.js";
+import { aggregation } from "../index.js";
+import * as tf from '@tensorflow/tfjs'
+
+
+const debug = createDebug("discojs:aggregator:mean");
+
+type ThresholdType = 'relative' | 'absolute'
+
+/** 
+ * Mean aggregator whose aggregation step consists in computing the mean of the received weights. 
+ * 
+ */
+export class ByzantineRobustAggregator extends Aggregator {
+    readonly #threshold: number;
+    readonly #thresholdType: ThresholdType;
+    private readonly clippingRadius: number;
+    private readonly maxIterations: number;
+    #minNbOfParticipants: number | undefined;
+    private momentumHistory: Map<client.NodeID, WeightsContainer> = Map()
+
+    /**
+     * Create a mean aggregator that averages all weight updates received when a specified threshold is met.
+     * By default, initializes an aggregator that waits for 100% of the nodes' contributions and that
+     * only accepts contributions from the current round (drops contributions from previous rounds).
+     * 
+     * @param threshold - how many contributions trigger an aggregation step.
+     * It can be relative (a proportion): 0 < t <= 1, requiring t * |nodes| contributions. 
+     * Important: to specify 100% of the nodes, set `threshold = 1` and `thresholdType = 'relative'`.
+     * It can be an absolute number, if t >=1 (then t has to be an integer), the aggregator waits fot t contributions
+     * Note, to specify waiting for a single contribution (such as a federated client only waiting for the server weight update),
+     * set `threshold = 1` and `thresholdType = 'absolute'`
+     * @param thresholdType 'relative' or 'absolute', defaults to 'relative'. Is only used to clarify the case when threshold = 1, 
+     * If `threshold != 1` then the specified thresholdType is ignored and overwritten
+     * If `thresholdType = 'absolute'` then `threshold = 1` means waiting for 1 contribution
+     * if `thresholdType = 'relative'` then `threshold = 1`` means 100% of this.nodes' contributions, 
+     * @param roundCutoff - from how many past rounds do we still accept contributions. 
+     * If 0 then only accept contributions from the current round, 
+     * if 1 then the current round and the previous one, etc.
+     */
+    constructor(roundCutoff = 0, threshold = 1, thresholdType?: ThresholdType, clippingRadius: number = 1.0, maxIterations: number = 10) {
```

Comment:
type trivially infered

```suggestion
    constructor(roundCutoff = 0, threshold = 1, thresholdType?: ThresholdType, clippingRadius = 1.0, maxIterations = 10) {
```